scm-webapp/src/main/java/sonia/scm/repository/RepositoryInitializer.java

```diff
@@ -77,17 +83,24 @@ public void initialize(Repository repository) {
 
     private final Repository repository;
     private final ModifyCommandBuilder builder;
+    private final Map<String, JsonNode> creationContext;
 
-    InitializerContextImpl(Repository repository, ModifyCommandBuilder builder) {
+    InitializerContextImpl(Repository repository, ModifyCommandBuilder builder, Map<String, JsonNode> creationContext) {
       this.repository = repository;
       this.builder = builder;
+      this.creationContext = creationContext;
     }
 
     @Override
     public Repository getRepository() {
       return repository;
     }
 
+    @Override
+    public Map<String, JsonNode> getCreationContext() {
+      return creationContext;
+    }
```

Comment:
As mentioned above an api which does the whole unmarshalling for bob would be much nicer.